### PR TITLE
Add user authentication and set up docker containers

### PR DESCRIPTION
**Pull request by bilalebi**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
```
Comment (nerdstrike) [inline LICENSE:182]: I believe it says "Don't include the brackets"...

File: NOTICE [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Ensembl
+Copyright 2018-2020 EMBL-European Bioinformatics Institute
```
Comment (nerdstrike) [inline NOTICE]: 2018 is an interesting starting year

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+asgiref==3.2.10
```
Comment (nerdstrike) [inline requirements.txt:1]: I'm not a big fan of fixing library versions until it becomes necessary to do so (mainly to stop long term incompatibilities cropping up when another dependency needs to be bumped). Your call.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+mysqlclient==2.0.1
+packaging==20.4
+pluggy==0.13.1
+py==1.9.0
+pyparsing==2.4.7
+pytest==6.0.1
+pytest-cov==2.10.1
+pytest-django==3.9.0
+pytz==2020.1
+six==1.15.0
```
Comment (nerdstrike) [inline requirements.txt:16]: Surely you don't need six for a pure Python 3 project?

File: thr/user/tests.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        kwargs['password'] = test_password
+        if 'username' not in kwargs:
+            kwargs['username'] = str(uuid.uuid4())
+        return django_user_model.objects.create_user(**kwargs)
+
+    return make_user
+
+
+@pytest.fixture
+def auto_login_user(db, client, create_user, test_password):
```
Comment (nerdstrike) [inline thr/user/tests.py:97]: Don't understand what this is testing. Please comment

**Review by nerdstrike — CHANGES_REQUESTED** [review]
A lack of Django familiarity means I can't usefully review the plumbing, but I've spotted some obvious things.

File: requirements.txt (hunk 3 above)
Comment (bilalebi) [reply requirements.txt:1]: I used `pip freeze > requirements.txt` which create a snapshot in time of the packages used their versions 
I think both approaches have their pros and cons, adding library's version guarantee that we won't end up in dependency hell.. some versions of packages will not work well with other packages and their respective versions. It is hard to predict how changes in the future will affect these interrelationships.

File: requirements.txt (hunk 4 above)
Comment (bilalebi) [reply requirements.txt:16]: For some reason, it's required by `packaging` which is required by `pytest`
```
$ pip show packaging    
Name: packaging
Version: 20.4
Summary: Core utilities for Python packages
Home-page: https://github.com/pypa/packaging
Author: Donald Stufft and individual contributors
Author-email: donald@stufft.io
License: BSD-2-Clause or Apache-2.0
Location: /Users/username/git/thr/thr_env/lib/python3.8/site-packages
Requires: six, pyparsing
Required-by: pytest
```

Comment (nerdstrike) [reply requirements.txt:16]: I suppose I'm surprised that pip doesn't take care of that.

Comment (bilalebi) [reply requirements.txt:16]: The `requirements.txt` itself is generated using pip 
Just to make sure `six` is needed I uninstalled it and run `pytest`, then I got the error:
```
File "/Users/username/git/thr/thr_env/lib/python3.8/site-packages/pytest_django/live_server_helper.py", line 1, in <module>
    import six
ModuleNotFoundError: No module named 'six'
```

File: thr/user/tests.py (hunk 5 above)
Comment (bilalebi) [reply thr/user/tests.py:97]: Done, let me if it's clear

File: LICENSE (hunk 1 above)
Comment (bilalebi) [reply LICENSE:182]: Yes.. it says _Don't include the brackets_, I don't see any bracket in the source code

File: requirements.txt (hunk 3 above)
Comment (nerdstrike) [reply requirements.txt:1]: Ah, now I remember why I hate it. You can't use a shared python env without it messing with other projects.

**Review by nerdstrike — APPROVED** [review]
All my superficial concerns are addressed. I recommend you wait for @sanjay-boddu 's approval before merging.

**Pull request merged**